tests/check-snapshots.sh

```diff
@@ -4,9 +4,16 @@ set -u
 
 readonly SHECC="$PWD/out/shecc"
 
+if [ "$#" != 1 ]; then
+    echo "Usage: $0 <backend>"
+    exit 1
+fi
+
+readonly BACKEND="$1"
```

Comment:
Rename to `ARCH` for consistency. "Backend" is not a proper term since architecture-specific defines must appear before C compilation.